### ISSUE TITLE
[Bug]: storybook cli init not working with NX "package based" repo

### ISSUE THREAD

**Issue by unleashit** [storybookjs/storybook#26079]
### Describe the bug

Running either `npx storybook@latest init` or `npx storybook@next init` in an NX monorepo produces the following error:

```bash
SB_CLI_INIT_0001 (NxProjectDetectedError): We have detected Nx in your project. Nx has its own Storybook initializer
, so please use it instead.
Run "nx g @nx/storybook:configuration" to add Storybook to your project.
```
Forcing with the -f flag doesn't work either.

This may be intentional in order to let NX manage versions of the same package with its generators. But NX has [several flavors](https://nx.dev/concepts/integrated-vs-package-based) of ways to work with it. In this project, I'm using "package based" which means generators are not used and each package has a normal package.json (in addition to a main one in the root). Workspaces are managed by n/pnpm or yarn. It's pretty similar to how Turborepo works or stand alone Lerna.

As far as I can tell, there is currently no way using the CLI to install Storybook in an NX repo without a generator. That means, I'll have to install it manually in this project since I'm not using generators (which isn't a big deal as long as I can!).

### To Reproduce

No repro, but here's the monorepo in question: https://github.com/unleashit/npm-library. Steps I've taken

1. npx create-vite ./demos/storybook --template react-ts
2. cd into storybook folder
3. npx storybook@latest init --force

### System

```bash
System:
    OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 20.5.0 - ~/.nvm/versions/node/v20.5.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v20.5.0/bin/yarn
    npm: 10.2.5 - ~/.nvm/versions/node/v20.5.0/bin/npm
    pnpm: 8.15.1 - ~/.nvm/versions/node/v20.5.0/bin/pnpm <----- active
  Browsers:
    Chrome: 121.0.6167.85
```

**Comment by loc** [comment]
I agree that this error is heavy-handed and should probably be improved, but FWIW @unleashit I was able to work around the same error with:

`yarn dlx storybook@latest init -t react -b vite`

**Comment by unleashit** [comment]
Thanks, nice find. I wonder if it's more about the flags being present though, since pnpx storybook@latest init -t react -b vite (no dlx) worked for me? Anyway, seems to be happy so far. We'll see what happens when I have more time and remove some of duplicated deps like Typescript, etc.

**Comment by geoquant** [comment]
why can't i eject from the nx ecosystem? this has to be a joke: 

   We have detected Nx in your project. Nx has its own Storybook initializer, so please use it instead.
        Run "nx g @nx/storybook:configuration <your-project-name>" to add Storybook to a given Nx app or lib.`,

**Comment by HughEdwards1** [comment]
The only reference I have to `nx` in my project is through a dev dependency, `@typescript-eslint/types/package.json`. 
Enough to tell me I'm using nx and the installer won't work.

Using the flags as above but this is an incredibly poor onboarding
